include/pybind11/pybind11.h

```diff
@@ -90,7 +90,7 @@ class cpp_function : public function {
     /// but with an added `&`.
     template <typename Return, typename Class, typename... Arg, typename... Extra>
     cpp_function(Return (Class::*f)(Arg...)&, const Extra&... extra) {
-        initialize([f](Class *c, Arg... args) -> Return { return (c->*f)(args...); },
+        initialize([f](Class *c, Arg... args) -> Return { return (c->*f)(std::forward<Arg>(args)...); },
```

Comment:
Shouldn't this be `Arg&&... args`, just like in `functional.h`?

Reply:
I agree and I actually tried to apply the perfect-forwarding pattern in these 4 class-method forwarders as well, but it failed terribly.

Reply:
I think, improving the perfect forwarding should be another PR. I reverted the corresponding commit as it isn't relevant for this PR.